codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxySSLTLV.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.haproxy;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static io.netty.util.internal.ObjectUtil.*;
+
+/**
+ * Represents a {@link HAProxyTLV} of the type {@link HAProxyTLV.Type#PP2_TYPE_SSL}.
+ * This TLV encapsulates other TLVs and has additional information like verification information and a client bitfield.
+ */
+public final class HAProxySSLTLV extends HAProxyTLV {
+
+    private final int verify;
+    private final List<HAProxyTLV> tlvs;
+    private final Set<CLIENT> clients;
+
+    /**
+     * The possible values of the clientfield bitmask
+     */
+    public enum CLIENT {
```

Comment:
I checked the Netty Source and most of the classes use upper-case enum values, only a few classes use lower case enum values. Since the HAProxy protocol also defines these enum values in upper case (see http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt), I believe it would be beneficial to have the values in upper case. If camel case enum values are mandatory I can change this of course.